### ISSUE TITLE
Monorepo and dependency installation check

### ISSUE THREAD

**Issue by npirotte** [remix-run/remix#2677]
### What version of Remix are you using?

remix@1.3.5-pre.1

### Steps to Reproduce

* Setup a mono repo (for example with nx) with a remix application, having it's own package.json, and at least another library that have some third party dependency (simply install and import lodash, that will be enough).
* From remix application, import the component from the other library (so the third party is included into the build)
* Build or dev the remix application

### Expected Behavior

App is built or dev mode is started, no warning.

### Actual Behavior

App is build or dev mode is started, application is running fine, but a warning is printed in the console for each dependency that are imported by some of the monorepo component but not by remix application itself.

example:
```
> npx remix build

Building Remix app in production mode...
The path "react-dom/server" is imported in app/entry.server.tsx but react-dom is not listed in your package.json dependencies. Did you forget to install it?
The path "react" is imported in app/routes/demos/actions.tsx but react is not listed in your package.json dependencies. Did you forget to install it?
The path "@react-hook/size" is imported in ../bar-chart/src/lib/bar-chart.tsx but @react-hook/size is not listed in your package.json dependencies. Did you forget to install it?
The path "@visx/scale" is imported in ../bar-chart/src/lib/bar-chart.tsx but @visx/scale is not listed in your package.json dependencies. Did you forget to install it?
The path "@ngneat/falso" is imported in ../bar-chart/src/lib/bar-chart.tsx but @ngneat/falso is not listed in your package.json dependencies. Did you forget to install it?
The path "@visx/group" is imported in ../bar-chart/src/lib/bar-chart.tsx but @visx/group is not listed in your package.json dependencies. Did you forget to install it?
The path "@visx/shape" is imported in ../bar-chart/src/lib/bar-chart.tsx but @visx/shape is not listed in your package.json dependencies. Did you forget to install it?
The path "lodash/fp" is imported in ../bar-chart/src/lib/bar-chart.tsx but lodash is not listed in your package.json dependencies. Did you forget to install it?
The path "@visx/axis" is imported in ../bar-chart/src/lib/bar-chart.tsx but @visx/axis is not listed in your package.json dependencies. Did you forget to install it?
Built in 357m
```

Here is a link to the repo that generate those logs (running nx build demo)

**Comment by F3n67u** [comment]
@npirotte could you create a reproduction repo for this issue?

**Comment by npirotte** [comment]
ha, I forgot to paste the link: https://github.com/npirotte/hello-visx

Just run `nx build demo`

**Comment by F3n67u** [comment]
remix doesn't know you specify some common those deps at the monorepo root `package.json`, so this is why remix complains: `dep not listed in your package.json dependencies`.

https://github.com/remix-run/remix/blob/5144a17eccb788f56ad58f5e0f122af1047631a7/packages/remix-dev/compiler/dependencies.ts#L15-L19

rootDirectory is `./packages/demo`

**Comment by F3n67u** [comment]
This warning is introduced by https://github.com/remix-run/remix/pull/2558/files#diff-ef8b0a35f03b499672160742c2d6b9688717e8d6b923dde0e10d2631ab0c09d6R419

**Comment by npirotte** [comment]
Would it be possible somehow that remix check parent project dependencies?

Or at least give the possibility to skip that check that can eventually be done by eslint for these more complex cases :)

**Comment by F3n67u** [comment]
> Would it be possible somehow that remix check parent project dependencies?
> 
> Or at least give the possibility to skip that check that can eventually be done by eslint for these more complex cases :)

@jacob-ebey what do you think?

**Comment by sergiocarneiro** [comment]
@F3n67u In a DX perspective, the best solution would be to set a path in `remix.config.js`:

```js
/**
 * @type {import('@remix-run/dev').AppConfig}
 */
module.exports = {
  cacheDirectory: "./node_modules/.cache/remix",
  ignoredRouteFiles: [".*", "**/*.css", "**/*.test.{js,jsx,ts,tsx}"],

  // link to monorepo root
  rootPath: "../../"
};
```

**Comment by npirotte** [comment]
@F3n67u to me indeed the best solution would be to leave those things up to a specialised tool like eslint https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md

Almost all the developers are using it (it might even be already turned on in company wide eslint config), so imo better let remix focus on its core topic: being an awesome framework to build web apps, and don't spend time side things like this :)

**Comment by sergiocarneiro** [comment]
> @F3n67u to me indeed the best solution would be to leave those things up to a specialised tool like eslint https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md

Makes sense if remix doesn't need to know about dependencies.

**Comment by F3n67u** [comment]
I filed a pr to solve this problem. so we have two option now:
1. merge my pr to solve the warning problem.
2. remove the warning logic and let eslint do analysis like @npirotte said

I am happy with either option 1 or option 2。

**Comment by dodas** [comment]
Thumbs up for removing the check and leaving it to eslint.

**Comment by MichaelDeBoey** [comment]
Closed by #3085
